lib/stringify.js

```diff
@@ -27,13 +29,15 @@ var defaults = {
     },
     skipNulls: false,
     strictNullHandling: false
+    // emptyValue: some flag
```

Comment:
I'm not sure what the best way to handle this case is, as there needs to way to recognize "user provided no option" so the default behavior of outputting `foo=` as opposed to `foo` can be used.

I've seen things like using the global `Infinity`, storing the option in an object and using `hasOwnProperty`, etc. among others but not sure what you'd prefer. Thanks!

Reply:
Yeah -- let's say we don't provide a default. This means when stringifying `{ foo: undefined }` we get `'foo'` whereas before we get `''` (line 77). If we do provide a default, stringifying `{ foo: [whatever our chosen default is]}` gives `'foo'` instead of whatever it gave before my change. So without some more involved code, stringifying whatever value we choose as the default would change from whatever it was before to `'foo'`.

Reply:
I'd like this not to be a breaking change, so `qs.stringify({ foo: undefined })` needs to return the same value in master as it does in this PR.

Reply:
cc @timhwang21, have you had any time to work on this PR?

Reply:
In all honesty, no, this entirely fell off my radar. Will catch up on the past few years of discussion 😅